src/test/java/org/example/bookshop/controller/BookControllerTest.java

```diff
@@ -0,0 +1,230 @@
+                    "database/categories/add-category-to-categories-table.sql",
+                    "database/books/add-book-to-books-table.sql",
+                    "database/categories/add-category-for-books_categories-table.sql"
+            );
+            for (String scriptPath : scriptPaths) {
+                ScriptUtils.executeSqlScript(connection, new ClassPathResource(scriptPath));
+            }
+        }
+    }
+
+    @Test
+    @Sql(scripts = "classpath:database/books/delete-all-books-from-books-table.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {"classpath:database/books/delete-all-books-from-books-table.sql",
+            "classpath:database/books/add-book-to-books-table.sql",
+            "classpath:database/categories/add-category-for-books_categories-table.sql"},
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @WithMockUser(roles = {"ADMIN"})
+    @DisplayName("Create a new book")
+    void createBook_ValidRequestDto_Success() throws Exception {
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+        MvcResult result = mockMvc.perform(
+                post("/books")
+                .content(jsonRequest)
+                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isCreated())
+                .andReturn();
+        BookDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), BookDto.class);
+
+        EqualsBuilder.reflectionEquals(bookDto, actual);
+    }
+
+    @Test
+    @Sql(scripts = {"classpath:database/books/add-two-books-to-books-table.sql",
+            "classpath:database/categories/add-two-category-to-books_categories-table.sql"},
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/books/delete-two-books-from-books-table.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @WithMockUser(roles = {"USER"})
+    @DisplayName("Get all books")
+    void getAllBooks_Success() throws Exception {
+        BookDto secondBookDto = new BookDto();
+        secondBookDto.setId(SECOND_ID);
+        secondBookDto.setTitle(BOOK_TITLE + UNIQUE_PARAM);
+        secondBookDto.setAuthor(BOOK_AUTHOR + UNIQUE_PARAM);
+        secondBookDto.setIsbn(BOOK_ISBN + UNIQUE_PARAM);
+        secondBookDto.setPrice(BOOK_PRICE);
+        secondBookDto.setCategoryIds(categoriesId);
+        BookDto thirdBookDto = new BookDto();
+        thirdBookDto.setId(THIRD_ID);
+        thirdBookDto.setTitle(UNIQUE_PARAM + BOOK_TITLE);
+        thirdBookDto.setAuthor(UNIQUE_PARAM + BOOK_AUTHOR);
+        thirdBookDto.setIsbn(UNIQUE_PARAM + BOOK_ISBN);
+        thirdBookDto.setPrice(BOOK_PRICE);
+        thirdBookDto.setCategoryIds(categoriesId);
+        List<BookDto> expected = List.of(bookDto, secondBookDto, thirdBookDto);
+
```

Comment:
Better move this logic to a private method or you can create a util class for generation data. 
I also recommend you use the construction Given - When - Then